Std/Data/Rat/Basic.lean

```diff
@@ -27,6 +28,9 @@ denominator is positive and the numerator and denominator are coprime.
   reduced : num.natAbs.Coprime den := by decide
   deriving DecidableEq
 
+theorem Rat.ext : {p q : Rat} → p.num = q.num → p.den = q.den → p = q
+  | ⟨_,_,_,_⟩, ⟨_,_,_,_⟩, rfl, rfl => rfl
+
```

Comment:
This should be in Rat.Lemmas.

It should also have `@[local ext]`. Applying the local attribute to an existing lemma still works. It's only the part that automatically generates the lemmas for a structure which is broken.

Reply:
I don't know if a local ext matters much.  I'm inclined to upstream this.  @adomani if you want to move it, that's great, but I can also just pull it in and change it in a small PR.

Reply:
@joehendrix you mean to move the lemma to `Rat.lemmas`?  I can do that right away, if that is what you wanted!

Reply:
@adomani Thanks, moving to Rat.Lemmas si what I meant.  Once that's moved, I'll merge.

Reply:
Great, thanks: I just pushed the move!